libsolidity/formal/SMTEncoder.cpp

```diff
@@ -1245,6 +1255,24 @@ void SMTEncoder::arrayAssignment()
 
 void SMTEncoder::indexOrMemberAssignment(Expression const& _expr, smtutil::Expression const& _rightHandSide)
 {
+	if (auto const* memberAccess = dynamic_cast<MemberAccess const*>(&_expr))
+	{
+		if (dynamic_cast<ContractDefinition const*>(expressionToDeclaration(memberAccess->expression())))
+		{
+			auto const* var = dynamic_cast<VariableDeclaration const*>(memberAccess->annotation().referencedDeclaration);
+			if (var)
```

Comment:
```suggestion
			if (auto const* var = dynamic_cast<VariableDeclaration const*>(memberAccess->annotation().referencedDeclaration))
```

Also, you can get rid of braces if you want